File: backend/src/main/java/com/police/kb/domain/entity/PromptTemplate.java
package com.police.kb.domain.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;

import java.io.Serializable;
import java.time.LocalDateTime;

@Data
@TableName("prompt_template")
public class PromptTemplate implements Serializable {

    private static final long serialVersionUID = 1L;

    @TableId(type = IdType.AUTO)
    private Long id;

    private String name;

    private String content;

    @TableField("`variables`")
    private String variables;

    private String description;

    @TableField("is_default")
    private Integer isDefault;

    @TableField("is_system")
    private Integer isSystem;  // 0-我的模板 1-系统模板 2-共享模板

    private Integer sort;

    @TableField("created_by")
    private Long createdBy;

    @TableField("created_time")
    private LocalDateTime createdTime;

    @TableField("updated_by")
    private Long updatedBy;

    @TableField("updated_time")
    private LocalDateTime updatedTime;

    @TableLogic
    @TableField("deleted")
    private Integer deleted;
}

File: frontend/src/api/promptTemplate.ts
import request from '@/utils/request'

export interface PromptTemplate {
  id: number
  name: string
  content: string
  variables: string
  description: string
  isDefault: number
  isSystem: number
  sort: number
  createdBy?: number
  createdTime?: string
  updatedBy?: number
  updatedTime?: string
}

export interface CreateTemplateRequest {
  name: string
  content: string
  variables?: string
  description?: string
  sort?: number
}

export enum TemplateType {
  SYSTEM = 'system',
  MY = 'my',
  SHARED = 'shared'
}

export function getTemplates(type: TemplateType = TemplateType.MY) {
  return request.get<PromptTemplate[]>('/v1/templates', {
    params: { type }
  })
}

export function getTemplate(id: number) {
  return request.get<PromptTemplate>(`/v1/templates/${id}`)
}

export function createTemplate(data: CreateTemplateRequest, isSystem: 0 | 1 | 2 = 0) {
  return request.post<number>('/v1/templates', { ...data, isSystem })
}

export function createSharedTemplate(data: CreateTemplateRequest) {
  return request.post<number>('/v1/templates', { ...data, isSystem: 2 })
}

export function updateTemplate(id: number, data: CreateTemplateRequest) {
  return request.put(`/v1/templates/${id}`, data)
}

export function deleteTemplate(id: number) {
  return request.delete(`/v1/templates/${id}`)
}

export function setDefault(id: number) {
  return request.put(`/v1/templates/${id}/default`)
}
